From 2763f34fa756e5688d3fb333dd253cc97aea3ab1 Mon Sep 17 00:00:00 2001
From: Jihoon Bang <jbang@nvidia.com>
Date: Fri, 5 Apr 2013 16:37:39 -0700
Subject: [PATCH 09/10] power: add camera hint

Bug 1261629

Change-Id: Ie02c9e4dfba59e24edc1a4937e0eaea3cf24ca6d
Reviewed-on: http://git-master/r/217030
(cherry picked from commit 073c6e237bb1c948ac38dfe708ae5d00faa3f733)
Reviewed-on: http://git-master/r/221644
Reviewed-by: Jihoon Bang <jbang@nvidia.com>
Tested-by: Jihoon Bang <jbang@nvidia.com>
Reviewed-by: Automatic_Commit_Validation_User
Reviewed-by: Dan Willemsen <dwillemsen@nvidia.com>
---
 include/hardware/power.h | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/include/hardware/power.h b/include/hardware/power.h
index baf3d6f..ae96aeb 100644
--- a/include/hardware/power.h
+++ b/include/hardware/power.h
@@ -47,7 +47,8 @@ typedef enum {
     POWER_HINT_HIGH_RES_VIDEO = 0x00000006,
     POWER_HINT_MIRACAST = 0x00000007,
     POWER_HINT_DISPLAY_ROTATION = 0x00000008,
-    POWER_HINT_COUNT = 0x00000009,
+    POWER_HINT_CAMERA = 0x00000009,
+    POWER_HINT_COUNT = 0x0000000A,
 } power_hint_t;
 
 /*
@@ -65,6 +66,18 @@ typedef enum {
     APP_PROFILE_COUNT,
 } app_profile_knob;
 
+/*
+ * Camera power hint enum, passed as data with POWER_HINT_CAMERA hint
+ */
+
+typedef enum {
+    CAMERA_HINT_POWER,
+    CAMERA_HINT_PERF,
+    CAMERA_HINT_FPS,
+    CAMERA_HINT_RESET,
+    CAMERA_HINT_COUNT,
+} camera_hint_t;
+
 /**
  * Every hardware module must have a data structure named HAL_MODULE_INFO_SYM
  * and the fields of this data structure must begin with hw_module_t
-- 
1.8.1.5

